edge-hub/src/Microsoft.Azure.Devices.Routing.Core/endpoints/StoringAsyncEndpointExecutor.cs

```diff
@@ -214,11 +222,12 @@ async Task SendMessagesPump()
                     {
                         await this.hasMessagesInQueue.WaitAsync(this.options.BatchTimeout);
 
-                        SortedDictionary<uint, EndpointExecutorFsm> snapshot = this.prioritiesToFsms;
+                        ImmutableDictionary<uint, EndpointExecutorFsm> snapshot = this.prioritiesToFsms;
                         bool haveMessagesRemaining = false;
 
+                        uint[] orderedPriorities = snapshot.Keys.OrderBy(k => k).ToArray();

```

Comment:
>uint[] orderedPriorities = snapshot.Keys.OrderBy(k => k).ToArray(); [](start = 24, length = 67)

see above comment, we're re-sorting here on inner loop, which seems unnecessary if the priorities haven't changed

Reply:
see reply above.